docs/modules/usage/how-to/custom-sandbox-guide.md

```diff
@@ -12,53 +12,131 @@ You have two options for customization:
 1. Use an existing image with the required software.
 2. Create your own custom Docker image.
 
-If you choose the first option, you can skip the `Create Your Docker Image` section.
+## Important Requirements
 
-## Create Your Docker Image
+Before proceeding with either option, note these key requirements:
+
+1. The base image must be Debian-based
+2. The user in the container MUST be `root` for proper functionality
+3. Any installed packages should be available system-wide, not just for specific users
 
-To create a custom Docker image, it must be Debian based.
+## Create Your Docker Image
 
-For example, if you want OpenHands to have `ruby` installed, create a `Dockerfile` with the following content:
+Here's a complete example of a custom runtime Dockerfile:
 
 ```dockerfile
 FROM debian:latest
 
 # Install required packages
-RUN apt-get update && apt-get install -y ruby
+RUN apt-get update && apt-get install -y \
+    ruby \
+    python3 \
+    nodejs \
+    && rm -rf /var/lib/apt/lists/*
+
+# Ensure we're running as root
+USER root
 ```
 
-Save this file in a folder. Then, build your Docker image (e.g., named custom-image) by navigating to the folder in
-the terminal and running::
+Save this file as `docker/runtime.Dockerfile`. Then, build your Docker image using one of these methods:
+
+### Method 1: Direct Docker Build
+
 ```bash
-docker build -t custom-image .
+docker build -t local/runtime:latest -f docker/runtime.Dockerfile .
 ```
 
-This will produce a new image called `custom-image`, which will be available in Docker.
+### Method 2: Using Docker Compose
+
+Create a `docker-compose.yml` file:
+
+```yaml
+services:
+  runtime:
+    build:
+      context: .
+      dockerfile: docker/runtime.Dockerfile
+    image: local/runtime:latest
+
+  openhands-app:
+    image: docker.all-hands.dev/all-hands-ai/openhands:0.14
+    environment:
+      - SANDBOX_RUNTIME_CONTAINER_IMAGE=local/runtime:latest
+      - LOG_ALL_EVENTS=true
+    volumes:
+      - /var/run/docker.sock:/var/run/docker.sock
+      - .cache:/home/openhands/.cache
+    ports:
+      - '3000:3000'
+    extra_hosts:
+      - "host.docker.internal:host-gateway"
+    depends_on:
+      - runtime
+```
 
-> Note that in the configuration described in this document, OpenHands will run as user "openhands" inside the
-> sandbox and thus all packages installed via the docker file should be available to all users on the system, not just root.
+Then run:
+```bash
+docker-compose up -d
+```
 
-## Using the Development Workflow
+## Configuration Options
 
-### Setup
+### Environment Variables
 
-First, ensure you can run OpenHands by following the instructions in [Development.md](https://github.com/All-Hands-AI/OpenHands/blob/main/Development.md).
+- `SANDBOX_RUNTIME_CONTAINER_IMAGE`: Specifies the custom runtime image (e.g., `local/runtime:latest`)
```

Comment:
Let's add a note here around `SANDBOX_BASE_CONTAINER_IMAGE` and the differences here.

Reply:
Agree, as a newcomer I find this confusing.